pkg/api/rest/delete.go

```diff
@@ -69,14 +69,13 @@ func BeforeDelete(strategy RESTDeleteStrategy, ctx api.Context, obj runtime.Obje
 		// only a shorter grace period may be provided by a user
 		if options.GracePeriodSeconds != nil {
 			period := int64(*options.GracePeriodSeconds)
-			if period > *objectMeta.DeletionGracePeriodSeconds {
-				return false, true, nil
+			newDeletionTimestamp := unversioned.NewTime(unversioned.Now().Add(time.Second * time.Duration(*options.GracePeriodSeconds)))
```

Comment:
We've discussed this with @wojtek-t and basically there are two ways of seeing that. My was the same as yours (that the second delete is basically an update on grace period). Wojteks was that it's the second, completely independent delete, and we should proceed with one that will happen sooner. After short consideration I think the latter is better.